File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = {id: 0 for id in id_list}
    hack = {id: [] for id in id_list}

    for case in report:
        a, b = case.split(' ')

        if a not in hack[b]:
            answer[b] += 1
            hack[b].append(a)

    result = [0] * len(id_list)
    for idx, user in enumerate(id_list):
        for reporter, reported_users in hack.items():
            if user in reported_users and len(reported_users) >= k:
                result[idx] += 1
                
    return result

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    const ans = new Array(id_list.length).fill(0)      
    // create report data list for each id
    const user = id_list.reduce((acc, id)=> ({...acc, [id]:[]}),{});  
    
    // checking report and push data into Object
    report.forEach((c)=>{
        const [a, b] = c.split(" ");
        if(!user[b].includes(a)) user[b].push(a); // push data to send a mail to reporter 
    }) 
     
    for (const u in user) {
        if (user[u].length >= k) {
          user[u].forEach((item) => (ans[id_list.indexOf(item)] += 1));
        }
    }
    return ans
}